compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/ExpectedTypeFinder.java

```diff
@@ -723,7 +751,12 @@ public Optional<TypeSymbol> transform(TableConstructorExpressionNode node) {
                         requireNonNullElse(((BTableType) referredType).constraint, expectedType)));
             }
 
-        } else if (isWithinParenthesis(linePosition, node.openBracket(), node.closeBracket())) {
+        }
+
+        Token openParen = node.openBracket();
```

Comment:
Since this is a bracket shall we use `openBracket` instead of `openParen`. Same for `closeParen`